hikari/polls.py

```diff
@@ -0,0 +1,318 @@
+    """The counts for each answer."""
+
+
+@attrs_extensions.with_copy
+@attrs.define(hash=False, kw_only=True, weakref_slot=False)
+class PollAnswerCount:
+    """Represents the count of a poll answer."""
+
+    answer_id: int = attrs.field(repr=True)
+    """The ID of the answer."""
+
+    count: int = attrs.field(repr=True)
+    """The number of votes for this answer."""
+
+    me_voted: bool = attrs.field(repr=True)
+    """Whether the current user voted for this answer."""
+
+
+class PollLayoutType(int, enums.Enum):
+    """Layout of a poll."""
+
+    DEFAULT = 1
+    """The default layout of a poll."""
+
+
+class PartialPoll:
+    """Base class for all poll objects."""
+
+    __slots__: typing.Sequence[str] = ("_question", "_answers", "_allow_multiselect", "_layout_type", "_counter")
+
+    def __init__(self, question: str, allow_multiselect: bool, layout_type: typing.Union[int, PollLayoutType]):
+        self._question = PollMedia(text=question)  # Only text is supported for question
+        self._allow_multiselect = allow_multiselect
+        self._layout_type = layout_type
+
+        # Answer is required, but we want users to user add_answer() instead of
+        # providing at initialization.
+        #
+        # Considering that answer ID can be arbitrary, `list`-based approaches
+        # like that of hikari.embeds.Embed._fields, while feasible to implement,
+        # would decrease long-term maintainability. I'm opting to use a `dict`
+        # here to simplify the implementation with some performance trade-off
+        # due to hashmap overhead.
+        self._answers: typing.MutableMapping[int, PollAnswer] = {}  # TODO: Do we need to set to None?
+
+    @property
+    def question(self) -> PollMedia:
+        """Returns the question of the poll."""
+        return self._question
+
+    @question.setter
+    def question(self, value: str) -> None:
+        self._question = PollMedia(text=value)
+
+    @property
+    def allow_multiselect(self) -> bool:
+        """Returns whether the poll allows multiple answers."""
+        return self._allow_multiselect
+
+    @allow_multiselect.setter
+    def allow_multiselect(self, value: bool) -> None:
+        self._allow_multiselect = value
+
+    @property
+    def layout_type(self) -> PollLayoutType:
+        """Returns the layout type of the poll."""
+        return PollLayoutType(self._layout_type)
+
+    @layout_type.setter
+    def layout_type(self, value: typing.Union[int, PollLayoutType]) -> None:
+        self._layout_type = value
+
+    @property
+    def answers(self) -> typing.MutableMapping[int, PollAnswer]:
+        """Returns the answers of the poll.
+
+        !!! note
+            Use [`hikari.polls.Poll.add_answer`][] to add a new answer,
+            [`hikari.polls.Poll.edit_answer`][] to edit an existing answer, or
+            [`hikari.polls.Poll.remove_answer`][] to remove a answer.
+        """
+        return self._answers
+
+
+class PollCreate(PartialPoll):
```

Comment:
Naming is something I dont like. We could easily modify the Poll object to allow instantiation, like Embed

Reply:
It's now `PollBuilder`